src/rgw/rgw_file.h

```diff
@@ -802,7 +802,10 @@ namespace rgw {
 			     const uint32_t flags = RGWFileHandle::FLAG_NONE) {
       using std::get;
 
-      LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
+      // This seems like a Clang compiler error.
+      // It result in irgw::RGWFileHandle::FLAG_NONE being undefined during linking.
+      // LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
+      LookupFHResult fhr { nullptr, 0 };
```

Comment:
thank you @wjwithagen ! by following the error message, i managed to create a reproducer and have a fix for it.
```c++
#include <cstdint>
#include <tuple>

struct Foo
{
  static constexpr uint32_t NONE = 0;
  typedef std::tuple<uint32_t> Tuple;
  void func() {
    Tuple t{NONE};
  }
};
  
int main()
{
  Foo().func();
}
```

because the constructor of `std::tuple<uint32_t>` always passes its parameters by reference if possible and the values of its parameters are evaluated, `Foo::NONE` is ODR-used[1]. but `Foo::NONE` out side of the class is not found anywhere. this fails the linker. so we can convert this lvalue to rvalue manually:

```diff
--- a/test.cc	2016-05-22 01:24:36.000000000 +0800
+++ b/test.cc	2016-05-22 01:24:43.000000000 +0800
@@ -7,7 +7,7 @@
   static constexpr uint32_t NONE = 0;
   typedef std::tuple<uint32_t> Tuple;
   void func() {
-    Tuple t{NONE};
+    Tuple t{uint32_t(NONE)};
   }
 };
```
so i don't think it is a "Clang compiler error". probably it's GCC who is too permissive.

@wjwithagen does this make sense to you? if so, maybe we can change our code in this way also to fix the issue.

---
[1] http://en.cppreference.com/w/cpp/language/definition